evm-contracts/test/v0.6/HistoricDeviationValidator.test.ts

```diff
@@ -0,0 +1,250 @@
+  personas = await setup.users(provider).then(x => x.personas)
+})
+
+describe('HistoricDeviationValidator', () => {
+  let validator: contract.Instance<HistoricDeviationValidatorFactory>
+  let flags: contract.Instance<FlagsFactory>
+  let ac: contract.Instance<SimpleWriteAccessControllerFactory>
+  const flaggingThreshold = 10000 // 10%
+  const previousRoundId = 2
+  const previousValue = 1000000
+  const currentRoundId = 3
+  const currentValue = 1000000
+
+  const deployment = setup.snapshot(provider, async () => {
+    ac = await acFactory.connect(personas.Carol).deploy()
+    flags = await flagsFactory.connect(personas.Carol).deploy(ac.address)
+    validator = await validatorFactory
+      .connect(personas.Carol)
+      .deploy(flags.address, flaggingThreshold)
+    await ac.connect(personas.Carol).addAccess(validator.address)
+  })
+
+  beforeEach(async () => {
+    await deployment()
+  })
+
+  it('has a limited public interface', () => {
+    matchers.publicAbi(validatorFactory, [
+      'THRESHOLD_MULTIPLIER',
+      'flaggingThreshold',
+      'flags',
+      'setFlagsAddress',
+      'setFlaggingThreshold',
+      'validate',
+      // Owned methods:
+      'acceptOwnership',
+      'owner',
+      'transferOwnership',
+    ])
+  })
+
+  describe('#constructor', () => {
+    it('sets the arguments passed in', async () => {
+      assert.equal(flags.address, await validator.flags())
+      matchers.bigNum(flaggingThreshold, await validator.flaggingThreshold())
+    })
+  })
+
+  describe('#validate', () => {
+    describe('when the deviation is greater than the threshold', () => {
+      const currentValue = 1100010
+
+      it('does raises a flag for the calling address', async () => {
+        const tx = await validator
+          .connect(personas.Nelly)
+          .validate(
+            previousRoundId,
+            previousValue,
+            currentRoundId,
+            currentValue,
+          )
+        const receipt = await tx.wait()
+        const event = matchers.eventExists(
+          receipt,
+          flags.interface.events.FlagOn,
+        )
+
+        assert.equal(flags.address, event.address)
+        assert.equal(
+          personas.Nelly.address,
+          h.evmWordToAddress(event.topics[1]),
+        )
+      })
+
+      it('uses less than the gas alotted by the aggregator', async () => {
+        const tx = await validator
+          .connect(personas.Nelly)
+          .validate(
+            previousRoundId,
+            previousValue,
+            currentRoundId,
+            currentValue,
+          )
+        const receipt = await tx.wait()
+        assert(receipt)
+        if (receipt && receipt.gasUsed) {
+          assert.isAbove(60000, receipt.gasUsed.toNumber())
+        }
+      })
+    })
+
+    describe('when the deviation is less than or equal to the threshold', () => {
+      const currentValue = 1100009
+
+      it('does raises a flag for the calling address', async () => {
```

Comment:
```suggestion
      it('does not raise a flag for the calling address', async () => {
```
Looks like this should be corrected.